src/model/events.js

```diff
@@ -6,19 +6,60 @@ export default class Events extends Observer {
     this._events = [];
   }
 
-  setEvents(events) {
+  setEvents(updateType, events) {
     this._events = events.slice();
+    this._notify(updateType);
   }
 
   getEvents() {
     return this._events;
   }
 
+  static adaptToClient(event) {
+    const adaptedEvent = Object.assign(
+        {},
+        event,
+        {
+          price: event.base_price,
+          isFavorite: event.is_favorite,
+          dateFrom: event.date_from,
+          dateTo: event.date_to
+        }
+    );
+
+    delete adaptedEvent.is_favorite;
+    delete adaptedEvent.base_price;
+    delete adaptedEvent.date_from;
+    delete adaptedEvent.date_to;
+
+    return adaptedEvent;
```

Comment:
Думаю, здесь стоит просто перечислить поля, которые нам нужны, явное объявление всегда лучше скрытого и получится гораздо короче:
```suggestion
    return {
      id: event.id,
      destination: event.destination,
      offers: event.offers,
      type: event.type,
      price: event.base_price,
      isFavorite: event.is_favorite,
      dateFrom: event.date_from,
      dateTo: event.date_to
    };
```

Reply:
поправил